src/main/java/org/mule/extension/http/api/request/builder/HttpRequesterSimpleRequestBuilder.java

```diff
@@ -105,9 +106,24 @@ protected void setRequestUriParams(Map<String, String> uriParams) {
     this.requestUriParams = uriParams;
   }
 
+  @Override
   public HttpRequestBuilder toHttpRequestBuilder(HttpRequesterConfig config) {
     return HttpRequest.builder(PRESERVE_HEADER_CASE || config.isPreserveHeadersCase())
         .headers(toMultiMap(getRequestHeaders()))
         .queryParams(toMultiMap(getRequestQueryParams()));
   }
+
+  public HttpRequesterRequestBuilder toHttpRequesterRequestBuilder() {
+    HttpRequesterRequestBuilder builder = new HttpRequesterRequestBuilder();
+    builder.setBody(TypedValue.of(requestBody));
+    builder.setUriParams(requestUriParams);
+    builder.setQueryParams(toMultiMap(requestQueryParams));
+    builder.setHeaders(toMultiMap(requestHeaders));
```

Comment:
we may need to do a perfromance check with this change, jusr in case

Reply:
This doesn't affect nothing that is already existing in the connector, but I agree that we should talk with performance team to create scenarios for this new source when we have the entire functionality implemented